### DISCUSSION TITLE
Make a note on documentation regarding the queryFulfilled

### DISCUSSION THREAD

**Discussion by panoramix360** [reduxjs/redux-toolkit#3077] [Ideas]
Just for context, I'm using the RTK Query on a React Native environment.

I have a code that uses the result from `queryFulfilled` to update a cached query.

```
async onQueryStarted(
        { parentId, profile },
        { dispatch, queryFulfilled },
      ) {
        try {
          const { data } = await queryFulfilled
          dispatch(
            api.util.updateQueryData(
              'getDatas',
              parentId,
              draft => {
                data.profile = profile
                draft.unshift(data)
              },
            ),
          )
        } catch {}
```

You can notice that the result returned by the `queryFulfilled` is being changed because the API does not return it.

While this works great on Debug environment, I noticed that the attribution was not working on a Release environment.

After some research, I discovered that React Native uses slightly different JavaScript runtimes for production and debugging.

So, to fix this I needed to change the code to copy the result and use this copied object to update the cached query.

```
async onQueryStarted(
        { parentId, profile },
        { dispatch, queryFulfilled },
      ) {
        try {
          const { data } = await queryFulfilled
          dispatch(
            api.util.updateQueryData(
              'getDatas',
              parentId,
              draft => {
                let dataCopy = Object.assign({}, data)
                dataCopy.profile = profile
                draft.unshift(dataCopy)
              },
            ),
          )
        } catch {}
```
And this finally worked locally and on production.

I don't know if I'm doing something wrong, but maybe this could be informed on the documentation like a warning to the devs.

Please, let me know what you think, and thank you for reading!

**Comment by markerikson** [comment]
I'm confused.  How does React Native's runtime affect behavior here?

**Reply by panoramix360** [reply]
yes, I was confused as well.

It seems to be related to the result object that is returned on `queryFulfilled`, in the debug runtime, the first solution worked with attribution:

`data.profile = profile`

But once I got to the production runtime, this same code didn't work as intended, property `profile` was not being settled by the attribution above.

It was only fixed when I copied the whole result object and then made the attribution, I don't know if this could be related to typescript as well, but it seems really strange to work on debug runtime and not on release.

**Reply by phryneas** [reply]
This might actually be something to raise on the `react-native` side of things as it looks like an error in their runtime or build process.

**Reply by panoramix360** [reply]
ok @phryneas thank you for your support!

it's really strange, I don't really know what is happening behind it, but it seems to be related to the different runtimes like it's described [here](https://reactnative.dev/docs/javascript-environment).
